File: FreelanceMarketplace/FreelanceMarketplace/GraphQL/Types/ApplyType.cs
using FreelanceMarketplace.Models;
using GraphQL.Types;

namespace FreelanceMarketplace.GraphQL.Types
{
    public class ApplyType : ObjectGraphType<Apply>
    {
        public ApplyType()
        {
            Field(x => x.ApplyId);
            Field(x => x.FreelancerId);
            Field(x => x.ClientId);
            Field(x => x.ProjectId);
            Field(x => x.Duration);
            Field(x => x.Status);
            Field(x => x.CreateAt);

            Field<UserType>("freelancer");
            Field<UserType>("client");
            Field<ProjectType>("project");
        }
    }
}

File: FE/src/types/ApplyType.ts
import { ProjectType } from "./ProjectType";
import { User } from "./UserType";

export interface ApplyType {
    applyId: number;
    projectId: number;
    freelancerId: number;
    clientId: number;
    duration: number;
    status: 'Pending' | 'Accepted' | 'Rejected';
    createAt: string;
    freelancer: User;
    client: User;
    project: ProjectType;
}

export interface ApplyInput {
    freelancerId: number;
    clientId: number;
    projectId: number;
    duration: number;
    status: 'Pending' | 'Accepted' | 'Rejected';
}